Passing LDFLAGS to libtool?

Hello!

I'm testing the static toolchain option. As I'm on Arch Linux I don't have any static libraries in the system. The first library that is needed is libfl (flex), so I downloaded the source, compiled it myself and installed into some local prefix. I then added the folder with this library to appropriate options:

```
[DEBUG]    CT_EXTRA_LDFLAGS_FOR_BUILD="-L${CT_TOP_DIR}/libs/lib"
[DEBUG]    CT_EXTRA_LDFLAGS_FOR_HOST="-L${CT_TOP_DIR}/libs/lib"
```

The build of binutils fails, and in build.log I see that these linker flags are not passed to libtool invocations for linking binutils:

```
[ALL  ]    libtool: link: x86_64-build_pc-linux-gnu-gcc -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow -I/home/freddie/bleeding-edge-toolchain-ct-ng/.build/src/binutils-2.25.1/binutils/../zlib -O2 -g -pipe -static-libstdc++ -static-libgcc -static -o bfdtest1 bfdtest1.o  ../bfd/.libs/libbfd.a -L/home/freddie/bleeding-edge-toolchain-ct-ng/.build/arm-none-eabi/build/build-binutils-host-x86_64-build_pc-linux-gnu/zlib -lz ../libiberty/libiberty.a -ldl
[ALL  ]    /usr/bin/ld: cannot find -lfl
```

On the other hand I see them used in other places, including some libtool invocations:

```
[ALL  ]    /usr/bin/bash ./libtool --silent --mode=link x86_64-build_pc-linux-gnu-gcc -I/home/freddie/bleeding-edge-toolchain-ct-ng/.build/src/expat-2.1.0/lib -I. -O2 -g -pipe  -Wall -Wmissing-prototypes -Wstrict-prototypes -fexceptions  -DHAVE_EXPAT_CONFIG_H -L/home/freddie/bleeding-edge-toolchain-ct-ng/libs/lib -o xmlwf/xmlwf xmlwf/xmlwf.o xmlwf/xmlfile.o xmlwf/codepage.o xmlwf/unixfilemap.o libexpat.la
```

Any idea how to solve that (if that is at all possible)?

--- a/scripts/build/binutils/binutils.sh
+++ b/scripts/build/binutils/binutils.sh
@@ -235,7 +235,7 @@ do_binutils_backend() {
         "${CT_BINUTILS_EXTRA_CONFIG_ARRAY[@]}"
 
     if [ "${static_build}" = "y" ]; then
-        extra_make_flags+=("LDFLAGS=-all-static")
+        extra_make_flags+=("LDFLAGS=${ldflags} -all-static")
         CT_DoLog EXTRA "Prepare binutils for static build"
         CT_DoExecLog ALL ${make} ${JOBSFLAGS} configure-host
     fi

